spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/BaseSnapshotUpdateSparkAction.java

```diff
@@ -20,20 +20,17 @@
 package org.apache.iceberg.spark.actions;
 
 import java.util.Map;
-import org.apache.iceberg.actions.SnapshotUpdate;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.spark.sql.SparkSession;
 
-abstract class BaseSnapshotUpdateSparkAction<ThisT, R>
-    extends BaseSparkAction<ThisT, R> implements SnapshotUpdate<ThisT, R> {
+abstract class BaseSnapshotUpdateSparkAction<ThisT> extends BaseSparkAction<ThisT> {
```

Comment:
I had to drop interfaces like `SnapshotUpdate` from parent actions so that `ThisT` can be an action class.